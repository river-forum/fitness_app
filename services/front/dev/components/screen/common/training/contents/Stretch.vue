<template>
  <div class="test">
    <InputDate />
    <InputTrainingName class="mt-12" />
    <Button button-inner-text="記録する" class="mx-auto mt-12" />
  </div>
</template>

<script>
import { defineComponent } from '@nuxtjs/composition-api'
import InputDate from '../../../../ui/form/InputDate.vue'
import InputTrainingName from '../../inputTrainingName/InputTrainingName.vue'
import Button from '../../../../ui/button/Button.vue'

export default defineComponent({
  components: {
    InputDate,
    InputTrainingName,
    Button,
  },
})
</script>
