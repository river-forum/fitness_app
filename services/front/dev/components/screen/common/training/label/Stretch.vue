<template>
  <TrainingLabelItem
    :active="active"
    current-color="#a1c556"
    @click.native="selectedListener"
  >
    <template slot="contents">ストレッチ</template>
  </TrainingLabelItem>
</template>
<script>
import { defineComponent, computed } from '@nuxtjs/composition-api'
import TrainingLabelItem from './TrainingLabelItem.vue'

const IDENTIFY = 'stretch'

export default defineComponent({
  components: {
    TrainingLabelItem,
  },
  props: {
    current: {
      type: String,
      required: true,
    },
  },
  setup(props, ctx) {
    const active = computed(() => props.current === IDENTIFY)
    function selectedListener() {
      ctx.emit('selected', IDENTIFY)
    }

    return {
      selectedListener,
      active,
    }
  },
})
</script>
