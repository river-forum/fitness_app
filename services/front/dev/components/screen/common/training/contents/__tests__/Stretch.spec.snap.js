import { mount } from '@vue/test-utils'
import Stretch from '../Stretch.vue'

it('Stretch Component renders correctly', () => {
  const app = mount(Stretch)

  expect(app.html()).toMatchSnapshot()
})
