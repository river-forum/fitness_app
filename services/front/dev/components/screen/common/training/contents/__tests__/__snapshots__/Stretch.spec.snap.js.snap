// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stretch Component renders correctly 1`] = `
"<div class=\\"test\\">
  <div class=\\"flex justify-between w-full pb-1 text-2xl border-b border-solid\\">
    <div class=\\"date-text\\">
      <p class=\\"text\\">日付</p>
    </div>
    <div class=\\"vdp-datepicker text-right date-input\\">
      <div class=\\"\\">
        <!----> <input type=\\"text\\" readonly=\\"readonly\\" autocomplete=\\"off\\">
        <!---->
      </div>
      <div class=\\"vdp-datepicker__calendar\\" style=\\"display: none;\\">
        <header><span class=\\"prev\\">&lt;</span> <span class=\\"day__month_btn up\\">Dec 2020</span> <span class=\\"next\\">&gt;</span></header>
        <div class=\\"\\"><span class=\\"cell day-header\\">Sun</span><span class=\\"cell day-header\\">Mon</span><span class=\\"cell day-header\\">Tue</span><span class=\\"cell day-header\\">Wed</span><span class=\\"cell day-header\\">Thu</span><span class=\\"cell day-header\\">Fri</span><span class=\\"cell day-header\\">Sat</span> <span class=\\"cell day blank\\"></span><span class=\\"cell day blank\\"></span><span class=\\"cell day\\">1</span><span class=\\"cell day\\">2</span><span class=\\"cell day\\">3</span><span class=\\"cell day\\">4</span><span class=\\"cell day weekend sat\\">5</span><span class=\\"cell day weekend sun\\">6</span><span class=\\"cell day\\">7</span><span class=\\"cell day\\">8</span><span class=\\"cell day\\">9</span><span class=\\"cell day\\">10</span><span class=\\"cell day\\">11</span><span class=\\"cell day weekend sat\\">12</span><span class=\\"cell day weekend sun\\">13</span><span class=\\"cell day\\">14</span><span class=\\"cell day\\">15</span><span class=\\"cell day\\">16</span><span class=\\"cell day\\">17</span><span class=\\"cell day\\">18</span><span class=\\"cell day weekend sat\\">19</span><span class=\\"cell day weekend sun\\">20</span><span class=\\"cell day\\">21</span><span class=\\"cell day\\">22</span><span class=\\"cell day\\">23</span><span class=\\"cell day\\">24</span><span class=\\"cell day\\">25</span><span class=\\"cell day weekend sat\\">26</span><span class=\\"cell day weekend sun\\">27</span><span class=\\"cell day\\">28</span><span class=\\"cell day\\">29</span><span class=\\"cell day\\">30</span><span class=\\"cell day today\\">31</span></div>
      </div>
      <div class=\\"vdp-datepicker__calendar\\" style=\\"display: none;\\">
        <header><span class=\\"prev\\">&lt;</span> <span class=\\"month__year_btn up\\">2020</span> <span class=\\"next\\">&gt;</span></header> <span class=\\"cell month\\">January</span><span class=\\"cell month\\">February</span><span class=\\"cell month\\">March</span><span class=\\"cell month\\">April</span><span class=\\"cell month\\">May</span><span class=\\"cell month\\">June</span><span class=\\"cell month\\">July</span><span class=\\"cell month\\">August</span><span class=\\"cell month\\">September</span><span class=\\"cell month\\">October</span><span class=\\"cell month\\">November</span><span class=\\"cell month\\">December</span>
      </div>
      <div class=\\"vdp-datepicker__calendar\\" style=\\"display: none;\\">
        <header><span class=\\"prev\\">&lt;</span> <span>2020 - 2029</span> <span class=\\"next\\">&gt;</span></header> <span class=\\"cell year\\">2020</span><span class=\\"cell year\\">2021</span><span class=\\"cell year\\">2022</span><span class=\\"cell year\\">2023</span><span class=\\"cell year\\">2024</span><span class=\\"cell year\\">2025</span><span class=\\"cell year\\">2026</span><span class=\\"cell year\\">2027</span><span class=\\"cell year\\">2028</span><span class=\\"cell year\\">2029</span>
      </div>
    </div>
  </div> <input type=\\"text\\" placeholder=\\"トレーニング名を入力\\" class=\\"w-full p-6 text-2xl bg-gray-300 mt-12\\"> <button class=\\"flex items-center justify-center w-full p-6 text-2xl border button mx-auto mt-12\\">
    記録する
  </button>
</div>"
`;
